fix/table-filter-dropdown-styles

Aunshon opened the pull request:
### All Submissions:

* [x] My code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)
* [x] My code satisfies feature requirements
* [x] My code is tested
* [x] My code passes the PHPCS tests
* [x] My code has proper inline documentation
* [x] I've included related pull request(s) (optional)
* [x] I've included developer documentation (optional)
* [x] I've added proper labels to this pull request

### Related Pull Request(s)
* https://github.com/getdokan/dokan-pro/pull/5091


### Closes 
* Closes https://github.com/getdokan/plugin-internal-tasks/issues/1105
* Closes https://github.com/getdokan/plugin-internal-tasks/issues/1109

### How to test the changes in this Pull Request:
* Test all the select and multi-select across dokan lite and pro.


### Changelog entry
*No needed*

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added custom title labels for selected values in select components
  * Introduced icon support with left/right positioning in select fields
  * Multi-select now displays values as a compact summary view

* **Improvements**
  * Enhanced dropdown positioning and portal rendering for better UI placement
  * Refined styling and layout of select components and date picker

* **Documentation**
  * Updated component usage examples
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Aunshon committed:
fix: remove `isClearable` prop from AsyncSelect components for consistency

Aunshon committed:
fix: avoid dropdown clipping by rendering menus in a portal with correct positioning

Aunshon committed:
feat: add `selectedTitle` prop to AsyncSelect, enabling custom prefixes for single selections

Aunshon committed:
feat: enhance multi-select UI with summary display, checkbox-style options, and customizable components

Aunshon committed:
fix: optimize multi-select rendering by hiding duplicate chips in summary view and cleaning unused class attributes

Aunshon committed:
fix: update AsyncSelect styles for improved multi-select experience and enhance value container display

Aunshon committed:
refactor: extract reusable AsyncSelect components into individual files for better modularity and maintainability

Aunshon committed:
fix: update AsyncSelect styles for improved selection visibility and consistency

Aunshon committed:
fix: update AsyncSelect styles for improved selection visibility and consistency

Aunshon committed:
fix: enhance SingleValue component to conditionally display prefix in selection label

Aunshon committed:
fix: improve DateRangePicker styles and enhance button configurations for better UI consistency

mrabbani commented on src/components/CouponAsyncSelect.tsx:
```diff
@@ -272,6 +272,7 @@ function CouponAsyncSelect( props: CouponAsyncSelectProps ) {
                 return results;
             } }
             instanceId={ `coupon-async-${ depsSignature }` }
+            selectedTitle={ __( 'Coupon', 'dokan' ) }
```
Pls check the text domain 👎

mrabbani commented on src/components/OrderAsyncSelect.tsx:
```diff
@@ -266,6 +266,7 @@ function OrderAsyncSelect( props: OrderAsyncSelectProps ) {
                 return results;
             } }
             instanceId={ `order-async-${ depsSignature }` }
+            selectedTitle={ __( 'Order', 'dokan' ) }
```
Text domain 👎

mrabbani commented on src/components/ProductAsyncSelect.tsx:
```diff
@@ -282,6 +282,7 @@ function ProductAsyncSelect( props: ProductAsyncSelectProps ) {
             } }
             // pass an instanceId too for good measure
             instanceId={ `product-async-${ depsSignature }` }
+            selectedTitle={ __( 'Product', 'dokan' ) }
```
Text domain 👎

mrabbani commented on src/components/VendorAsyncSelect.tsx:
```diff
@@ -268,6 +268,7 @@ function VendorAsyncSelect( props: VendorAsyncSelectProps ) {
                 return results;
             } }
             instanceId={ `vendor-async-${ depsSignature }` }
+            selectedTitle={ __( 'Vendor', 'dokan' ) }
```
Text domain 👎

mrabbani commented on src/components/select/ValueContainer.tsx:
```diff
@@ -0,0 +1,39 @@
+    const prefix =
+        typeof titlePrefix === 'function' && labels.length
+            ? titlePrefix( values[ 0 ] )
+            : titlePrefix;
```
Pls review this code section which may lead to generate error.

mrabbani commented on src/components/select/styles.ts:
```diff
@@ -0,0 +1,74 @@
+        ...base,
+        zIndex: 9999,
+        wordBreak: 'break-all',
+        width: '18.75rem !important',
```
- May we use min-width?
- May we remove the ` !important'`?

mrabbani reviewed (changes requested):
- Almost all component do not have any props type. Pls try to introduce props type.
- Check the logic of all new component for the computed data.
- Be careful to use the text-domain for string translation.

Aunshon committed:
fix: simplify `selectedTitle` prop handling in AsyncSelect and update related components for consistency

- Removed function type support from `selectedTitle` prop.
- Updated related components to handle `selectedTitle` as a straightforward string.
- Standardized text-domain from `dokan` to `dokan-lite`.
- Adjusted `styles.ts` to use `minWidth` instead of `width` for better responsiveness.
- Enabled `isMulti` support in the withdraw dashboard page.

Aunshon committed:
fix: remove unused `isMulti` prop from AsyncSelect in withdraw dashboard page
